From c2f755626ca43be6c2b04af5f73e70aa8b7e4df1 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Sun, 4 Oct 2020 14:25:36 +0300
Subject: [PATCH 26/38] PCIe: imx6: imx7d: add support for phy refclk source
 configuration

i.MX7D variant of the IP can use either Crystal Oscillator input
or internal clock input as a Reference Clock input for PCIe PHY.
Make use of the optional property 'ext_osc' to configure the
reference clock.

Verified on Compulab SBC-iMX7 Single Board Computer.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 drivers/pci/controller/dwc/pci-imx6.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index eb21b5cbfab2..1fc6ef602dd1 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1634,7 +1634,8 @@ static void imx6_pcie_init_phy(struct imx6_pcie *imx6_pcie)
 	case IMX7D:
 	case IMX7D_EP:
 		regmap_update_bits(imx6_pcie->iomuxc_gpr, IOMUXC_GPR12,
-				   IMX7D_GPR12_PCIE_PHY_REFCLK_SEL, 0);
+				   IMX7D_GPR12_PCIE_PHY_REFCLK_SEL,
+				   imx6_pcie->ext_osc ? 0: IMX7D_GPR12_PCIE_PHY_REFCLK_SEL);
 		break;
 	case IMX6SX:
 	case IMX6SX_EP:
-- 
2.20.1

